bases/rsptx/assignment_server_api/assignment_builder/src/utils/sanitize.ts

```diff
@@ -0,0 +1,36 @@
+export const sanitizeId = (name: string, fallback?: string): string => {
+  const nameToUse = name || fallback || `generated_${Date.now()}`;
+
+  let sanitized = nameToUse
+    .replace(/[^a-zA-Z0-9\-_]/g, "_") // Replace invalid characters (including spaces) with underscore
+    .replace(/_+/g, "_") // Collapse multiple underscores to single
+    .replace(/^_+|_+$/g, ""); // Remove leading and trailing underscores
```

Comment:
Many questions from a pretext book will have a name that contain multiple underscores.  For example: `csawesome2__foo_bar-untit-3`

If the author included edition numbers then the name would be `csawesome2_2_foo_bar-unit-3` where the 2 indicates a second edition.